Feature/#28 푸시 알림 구현

Youngini opened the pull request:
## #️⃣연관된 이슈

> ex) #28

## 📝작업 내용

> FCM 사용을 위한 밑작업을 했습니다

## 💬리뷰 요구사항(선택)

> token 저장을 어떻게 할지 제대로 못정해서 관련 내용은 뺐습니다
> FCM을 사용하는데 파일 명들을 모두 FCM과 관련한 것으로 수정하는게 맞을까요?

Youngini committed:
build : firebase FCM 사용을 위한 설정 완료

Youngini committed:
build : firebase FCM 사용을 위한 설정 재설정

Youngini committed:
build : 잘못된 dependency들 삭제

Youngini committed:
build : Firebase ADMIN SDK 사용하기 위한 의존성 추가

Youngini committed:
feat : FirebaseConfig 파일 생성

Youngini committed:
refactor : User 정보 가져오기 위해 public으로 전환

Youngini committed:
feat : token을 받아오기 위한 dto 작성

Youngini committed:
feat : token을 받아오고 저장 및 메시지 보내는 code 작성

Youngini committed:
feat : token을 받아오고 저장하는 api 작성

Youngini committed:
feat : 푸시 알림 전송 api 구현

Youngini committed:
docs : controller 에 대한 문서 작성 완료

Youngini committed:
feat : FCM interface 구현

doyooon commented on src/main/java/io/github/eappezo/soundary/services/user/application/service/UserService.java:
Friend 에서도 getUser 쓰는데 그냥 이거 core로 올리는건 어떻게 생각하시나요

doyooon reviewed (commented):
LGTM

Youngini committed:
feat : FirebaseConfig errorcode 수정

Youngini committed:
feat : Firebase 실행 오류 추가

Youngini committed:
feat : 사용하지 않는 코드 삭제

Youngini committed:
feat : Entity 설정 완료

Youngini committed:
feat : FCM Service code 작성

Youngini committed:
feat : 팝업 알림을 위한 notification 정의

Youngini committed:
feat : UserDeviceRepository 생성

Youngini committed:
feat : FcmToken 생성을 위한 dto 설정

Youngini committed:
feat : FCM 토큰 발급 및 삭제 service code

Youngini committed:
chore : FCM key 숨기기

Youngini committed:
Merge branch 'Weekly7' of https://github.com/kakao-tech-campus-2nd-step3/Team22_BE into Feature/#28

GyuminBae commented on .gitignore:
```diff
@@ -41,4 +41,6 @@ bin/
\ No newline at end of file
+.DS_Store
+
+src/main/resources
```
resources 통째로 ignore하기 보단 특정 파일만 지정하거나 확장자로 ignore해주세요

Youngini committed:
feat : 사용하지 않는 코드 삭제

Youngini committed:
chore : 파일 구조 변경

Youngini committed:
chore : 숨길 파일 명 추가

Youngini committed:
feat : device 등록 추가

doyooon commented on src/main/java/io/github/eappezo/soundary/advice/notification/FCMNotificationWrapper.java:
이거 그냥 "" return 하는데 왜 잘 되나요? 진짜 궁금해

Youngini replied:
와 이거 제가 수정하고 커밋을 안올렸어요 수정본 커밋하겠습니다!

doyooon reviewed (commented):
LGTM

Youngini committed:
refactor : message 내 값들 return

getNotification()이 Message 클래스에서 public이 아니어서 접근이 불가능해서, Message 객체를 만들 때 알림 관련 데이터를 FCMNotificationWrapper에 전달하여 저장하고 사용하도록 수정
